server/controllers/reportController.js

```diff
@@ -0,0 +1,142 @@
+import sequelize from 'sequelize';
+import models from '../db/models';
+import utils from '../helpers/Utilities';
+
+
+const { Op } = sequelize;
+/**
+ * @Module ArticleController
+ * @description Controlls all activities related to Articles
+ */
+class reportController {
+  /**
+   * @description Creates an Article
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @returns {Object} Article data
+   * @memberof ArticleController
+   */
+  static async reportArticle(req, res) {
+    const article = await models.Article.findOne({
+      where: {
+        slug: req.params.slug
+      },
+      include: [{
+        model: models.User,
+        as: 'author',
+        attributes: ['firstname', 'lastname', 'username', 'image', 'email']
+      }]
+    });
+    if (!article) {
+      return utils.errorStat(res, 404, 'Article not found');
+    }
+
+    if (article) {
+      const authorId = req.user.id;
+      const articleId = article.id;
+      const { reportDetails, reportType } = req.body;
+      const findReport = await models.Report.findOne({
+        where: { authorId, articleId }
+      });
+
+      if (!findReport) {
+        const report = await models.Report.create({
+          reportDetails,
+          reportType,
+          authorId,
+          articleId
+        });
+        if (report) {
+          return utils.successStat(res, 201, 'report', report);
+        }
+      }
+      return utils.errorStat(res, 409, 'article already reported by you');
+    }
+  }
+
+  /**
+   * @description Creates an Article
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @returns {Object} Article data
+   * @memberof ReportController
+   */
+  static async getAllArticleReports(req, res) {
+    const allReports = await models.Report.findAll({
+      include: [
+        {
+          model: models.User,
+          as: 'author',
+          attributes: [
+            'firstname',
+            'lastname',
+            'username',
+            'image',
+            'email'
+          ]
+        },
+        {
+          model: models.Article,
+          attributes: [
+            'title',
+            'description',
+            'articleBody',
+            'image',
+            'tagList',
+            'slug'
+          ]
+        }
+      ]
+    });
+    return utils.successStat(res, 200, 'reports', allReports);
+  }
+
+  /**
+   * @description Creates an Article
+   * @param {Object} req - request object
+   * @param {Object} res - response object
+   * @returns {Object} Article data
+   * @memberof ReportController
+   */
+  static async getOneReport(req, res) {
+    const a = req.params.reportId;
+    const r = /^[0-9]*$/.test(a);
```

Comment:
Is there a reason why you're doing your param validation in the controller?

Reply:
It's a small check, and it would be an overkill to create a whole new file just for it

Reply:
I don't think you need a whole new file. You can just write a schema for it. What do you think?

Reply:
oh. nice. I'll do that